### PR TITLE
frontend (onnx): implementation graph visiting

### PR DESCRIPTION
**Pull request by alecerio**
Implementation of graph visiting:

- visit_graph
- build_module

And related tests:

- test_visit_graph_add
- test_visit_graph_sub
- test_build_module

@superlopuh @compor

### CODE REVIEW

File: xdsl/frontend/onnx/ir_builder.py [hunk 1]
```diff
@@ -52,3 +53,43 @@ def visit_value_info(i: ValueInfoProto, ctx: OnnxXdslMapping) -> Attribute:
...
     return t
+
+
+def build_module(graph: GraphProto) -> ModuleOp:
+    """Create the ModuleOp based on the onnx graph provided."""
+    module = ModuleOp([])
+
+    ctx = OnnxXdslMapping()
+    fn = visit_graph(graph, ctx)
+    module.regions[0].block.add_op(fn)
```
Comment (superlopuh) [inline xdsl/frontend/onnx/ir_builder.py:64]:
```suggestion
    ctx = OnnxXdslMapping()
    fn = visit_graph(graph, ctx)
    module = ModuleOp([fn])
```
Seems like we can simplify further now that the inner function returns a func op

File: xdsl/frontend/onnx/ir_builder.py [hunk 2]
```diff
@@ -52,3 +53,43 @@ def visit_value_info(i: ValueInfoProto, ctx: OnnxXdslMapping) -> Attribute:
...
+
+    return module
+
+
+def visit_graph(g: GraphProto, ctx: OnnxXdslMapping) -> IRDLOperation:
+    """
+    Visit the onnx graph to update the onnx context.
+    The nodes of the graph are visited only if all the inputs have already been visited.
+    When the node is visited, the associated state switches from 0 to 2.
```
Comment (superlopuh) [inline xdsl/frontend/onnx/ir_builder.py:73]: this seems outdated

**Review by superlopuh — APPROVED** [review]
LGTM

File: xdsl/frontend/onnx/ir_builder.py (hunk 2 above)
Comment (alecerio) [reply xdsl/frontend/onnx/ir_builder.py:73]: Done

File: xdsl/frontend/onnx/ir_builder.py (hunk 1 above)
Comment (alecerio) [reply xdsl/frontend/onnx/ir_builder.py]: Done

**Pull request merged**